BoxContentSDK/BoxContentSDK/BOXURLSessionCacheClient.m

```
@@ -39,8 +38,8 @@ - (id)initWithCacheRootDir:(NSString *)cacheRootDir
             NSString *cacheDir = [cacheRootDir stringByAppendingPathComponent:BOXURLSessionTaskCacheDirectoryName];
             if ([[NSFileManager defaultManager] fileExistsAtPath:cacheDir isDirectory:&isDir] == NO || isDir == NO) {
                 NSError *error = nil;
-                [[NSFileManager defaultManager] createDirectoryAtPath:cacheDir withIntermediateDirectories:YES attributes:nil error:&error];
-                BOXAssert(error != nil, @"Failed to create cacheDir %@ with error %@", cacheDir, error);
+                BOOL success = [[NSFileManager defaultManager] createDirectoryAtPath:cacheDir withIntermediateDirectories:YES attributes:nil error:&error];
+                BOXAssert(success, @"Failed to create cacheDir %@ with error %@", cacheDir, error);
```

What's the benefit of asserting here for developers? Should we just return nil if it silently fails?

---

yeah, assert to terminate in development mode. We will retry to create this dir when building subdir structures later. Do you think we should return nil and allow error as an argument for init so the user is aware of the issue?